ComfortZoneApp/app/src/main/java/com/example/comfortzone/utils/ComfortCalcUtil.java

```diff
@@ -171,8 +173,45 @@ private static void filterLowerTemps(ArrayList<ComfortLevelEntry> entryArrayList
         }
     }
 
-    private static void fillInTheBlanks() {
+    private static void findBlanks() {
         int countBlanks = 0;
+        int lowerIndex = 0;
+        int higherIndex = 0;
+        while (averages[higherIndex] == MIN_TEMP) {
+            higherIndex++;
+        }
+
+        if (higherIndex != lowerIndex) {
+            averages[lowerIndex] = averages[higherIndex] - (INTERVAL_SCALE * (higherIndex - lowerIndex));
+            fillInTheBlanks(lowerIndex, higherIndex);
+            lowerIndex = higherIndex;
+        }
+
+        for (int i = lowerIndex; i < TOTAL_LEVELS; i ++) {
+            if (averages[i] == MIN_TEMP) {
+                countBlanks++;
+            } else {
+                if (countBlanks > 0) {
+                    higherIndex = i;
+
+                    fillInTheBlanks(lowerIndex, higherIndex);
+                    lowerIndex = higherIndex;
 
+                } else {
+                    lowerIndex = i;
+                }
+            }
+        }
+    }
+
+    private static void fillInTheBlanks(int lowerIndex, int higherIndex) {
```

Comment:
Can rename this function to have a clearer name as well like fillInTheBlanksForAveragesOfLevelsTracker